spec/controllers/service_provider_controller_spec.rb

```diff
@@ -0,0 +1,52 @@
+require 'rails_helper'
+
+describe ServiceProviderController do
+  include SamlAuthHelper
+
+  describe '#update' do
+    let(:dashboard_sp_issuer) { 'some-dashboard-service-provider' }
+    let(:array_of_sps) do
+      [
+        {
+          issuer: dashboard_sp_issuer,
+          agency: 'a service provider',
+          friendly_name: 'a friendly service provider',
+          description: 'user friendly login.gov dashboard',
+          acs_url: 'http://sp.example.org/saml/login',
+          assertion_consumer_logout_service_url: 'http://sp.example.org/saml/logout',
+          block_encryption: 'aes256-cbc',
+          cert: saml_test_sp_cert
+        }
+      ]
+    end
+
+    context 'feature on' do
+      before do
+        SERVICE_PROVIDERS.delete dashboard_sp_issuer # start fresh
```

Comment:
What about using `stub_const`? It automatically resets at the end of the example

```
stub_const('SERVICE_PROVIDERS', [])
```